pkg/service/grpc_service.go

```diff
@@ -0,0 +1,129 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/open-feature/flagd/pkg/eval"
+	"github.com/open-feature/flagd/pkg/model"
+	gen "github.com/open-feature/flagd/schemas/protobuf/gen/v1"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+type GRPCServiceConfiguration struct {
+	Port int32
+}
+
+type GRPCService struct {
+	GRPCServiceConfiguration *GRPCServiceConfiguration
+	eval                     eval.IEvaluator
+	gen.UnimplementedServiceServer
+}
+
+func (s *GRPCService) SetEval(eval eval.IEvaluator) {
+	s.eval = eval
```

Comment:
Why not set this on a field, even public is more idiomatic than this?

Reply:
opened a new PR for this change 

https://github.com/open-feature/flagd/pull/75